src/cargo/util/toml/mod.rs

```diff
@@ -2442,16 +2456,34 @@ fn default_readme_from_package_root(package_root: &Path) -> Option<String> {
 
 /// Checks a list of build targets, and ensures the target names are unique within a vector.
 /// If not, the name of the offending build target is returned.
-fn unique_build_targets(targets: &[Target], package_root: &Path) -> Result<(), String> {
-    let mut seen = HashSet::new();
+fn unique_build_targets(
+    targets: &[Target],
+    package_root: &Path,
+) -> Result<(), HashMap<PathBuf, Vec<Target>>> {
+    let mut source_targets = HashMap::new();
     for target in targets {
         if let TargetSourcePath::Path(path) = target.src_path() {
             let full = package_root.join(path);
-            if !seen.insert(full.clone()) {
-                return Err(full.display().to_string());
+            match source_targets.entry(full) {
+                Entry::Vacant(e) => {
+                    e.insert(vec![target.clone()]);
+                }
+                Entry::Occupied(mut e) => {
+                    e.get_mut().push(target.clone());
+                }
```

Comment:
(nit)


```suggestion
            source_targets.entry(full).or_default().push(target.clone());
```

You also need to change the line where `source_targets` is defined to `let mut source_targets = HashMap::<_, Vec<_>>::new();` or else.

Reply:
Addressed! 
Thanks for your review!  💚 💙 💜 💛 ❤️